Python Updates 2022-07-15

mweinelt opened the pull request:
Second python update run for the big changes and some necessary stabilization work.

Hydra project: https://hydra.nixos.org/jobset/nixpkgs/python-updates
Regressions: https://malob.github.io/nix-review-tools-reports/
Previous run: https://github.com/NixOS/nixpkgs/pull/175805

mweinelt committed:
python3Packages.python-gvm: 21.11.0 -> 22.6.1

mweinelt committed:
python3Packages.python-homewizard-energy: 1.0.3 -> 1.1.0

mweinelt committed:
python3Packages.python-heatclient: 2.5.1 -> 3.0.0

mweinelt committed:
python3Packages.python-codon-tables: 0.1.11 -> 0.1.12

mweinelt committed:
python3Packages.pytest-xprocess: 0.18.1 -> 0.19.0

mweinelt committed:
python3Packages.python-dbusmock: 0.26.1 -> 0.28.1

mweinelt committed:
python3Packages.python-gnupg: 0.4.8 -> 0.4.9

mweinelt committed:
python3Packages.pytorch-lightning: 1.6.4 -> 1.6.5

mweinelt committed:
python3Packages.pytools: 2022.1.2 -> 2022.1.12

mweinelt committed:
python3Packages.python-json-logger: 2.0.2 -> 2.0.4

mweinelt committed:
python3Packages.python-lsp-server: 1.4.1 -> 1.5.0

mweinelt committed:
python3Packages.python-pam: 1.8.4 -> 2.0.2

mweinelt committed:
python3Packages.pytradfri: 9.0.0 -> 11.0.0

mweinelt committed:
python3Packages.pytorch-metric-learning: 1.2.0 -> 1.5.0

mweinelt committed:
python3Packages.qiskit: 0.36.1 -> 0.37.0

mweinelt committed:
python3Packages.qiskit-ibmq-provider: 0.19.1 -> 0.19.2

mweinelt committed:
python3Packages.qcs-api-client: 0.20.17 -> 0.21.0

mweinelt committed:
python3Packages.pyvisa: 1.11.3 -> 1.12.0

mweinelt committed:
python3Packages.pyzmq: 23.0.0 -> 23.2.0

mweinelt committed:
python3Packages.qdldl: 0.1.5.post0 -> 0.1.5.post2

mweinelt committed:
python3Packages.pyvisa-py: 0.5.1 -> 0.5.3

mweinelt committed:
python3Packages.pyturbojpeg: 1.6.6 -> 1.6.7

mweinelt committed:
python3Packages.qdarkstyle: 3.0.3 -> 3.1

mweinelt committed:
python3Packages.qiskit-ignis: 0.7.0 -> 0.7.1

mweinelt committed:
python3Packages.redis: 4.2.1 -> 4.3.4

mweinelt committed:
python3Packages.qnapstats: 0.4.0 -> 0.5.0

mweinelt committed:
python3Packages.QtPy: 2.0.1 -> 2.1.0

mweinelt committed:
python3Packages.rapidfuzz: 2.1.2 -> 2.1.3

mweinelt committed:
python3Packages.qiskit-nature: 0.3.2 -> 0.4.3

mweinelt committed:
python3Packages.readthedocs-sphinx-ext: 2.1.5 -> 2.1.8

mweinelt committed:
python3Packages.raincloudy: 1.1.1 -> 1.2.0

mweinelt committed:
python3Packages.rasterio: 1.2.10 -> 1.3.0

mweinelt committed:
python3Packages.qtconsole: 5.3.0 -> 5.3.1

mweinelt committed:
python3Packages.rpcq: 3.9.2 -> 3.10.0

mweinelt committed:
python3Packages.rtslib: 2.1.74 -> 2.1.75

mweinelt committed:
python3Packages.reolink: 0.60 -> 0.61

mweinelt committed:
python3Packages.rpy2: 3.4.5 -> 3.5.2

mweinelt committed:
python3Packages.repoze.sphinx.autointerface: 0.8 -> 1.0.0

mweinelt committed:
python3Packages.robotsuite: 2.2.1 -> 2.3.1

mweinelt committed:
python3Packages.reportlab: 3.6.9 -> 3.6.11

mweinelt committed:
python3Packages.robotframework-requests: 0.9.2 -> 0.9.3

mweinelt committed:
python3Packages.scikits.odes: 2.6.3 -> 2.6.4

mweinelt committed:
python3Packages.schema-salad: 8.2.20220204150214 -> 8.3.20220626185350

mweinelt committed:
python3Packages.scikit-hep-testdata: 0.4.12 -> 0.4.15

mweinelt committed:
python3Packages.sabyenc3: 5.4.2 -> 5.4.3

mweinelt committed:
python3Packages.selenium: 4.2.0 -> 4.3.0

mweinelt committed:
python3Packages.safety: 2.0.0 -> 2.1.0

mweinelt committed:
python3Packages.rx: 3.2.0 -> 4.0.4

mweinelt committed:
python3Packages.sigtools: 2.0.3 -> 4.0.0

mweinelt committed:
python3Packages.serpent: 1.40 -> 1.41

mweinelt committed:
python3Packages.setproctitle: 1.2.2 -> 1.2.3

mweinelt committed:
python3Packages.shtab: 1.5.4 -> 1.5.5

mweinelt committed:
python3Packages.setuptools-scm: 6.4.2 -> 7.0.5

mweinelt committed:
python3Packages.shap: 0.40.0 -> 0.41.0

mweinelt committed:
python3Packages.sip: 6.5.1 -> 6.6.2

mweinelt committed:
python3Packages.spacy-pkuseg: 0.0.28 -> 0.0.31

mweinelt committed:
python3Packages.sphinxcontrib-plantuml: 0.23 -> 0.24

mweinelt committed:
python3Packages.spacy: 3.3.1 -> 3.4.0

mweinelt committed:
python3Packages.soupsieve: 2.3.1 -> 2.3.2.post1

mweinelt committed:
python3Packages.solax: 0.2.9 -> 0.2.10

mweinelt committed:
python3Packages.sphinx: 4.5.0 -> 5.0.2

mweinelt committed:
python3Packages.spyder: 5.3.0 -> 5.3.2

mweinelt committed:
python3Packages.tabulate: 0.8.9 -> 0.8.10

mweinelt committed:
python3Packages.stevedore: 3.5.0 -> 4.0.0

mweinelt committed:
python3Packages.superqt: 0.3.2 -> 0.3.3

mweinelt committed:
python3Packages.tableaudocumentapi: 0.9 -> 0.10

mweinelt committed:
python3Packages.statmake: 0.4.1 -> 0.5.1

mweinelt committed:
python3Packages.spyder-kernels: 2.3.0 -> 2.3.2

mweinelt committed:
python3Packages.stanza: 1.3.0 -> 1.4.0

mweinelt committed:
python3Packages.starlette: 0.20.1 -> 0.20.4

mweinelt committed:
python3Packages.sqlalchemy-utils: 0.38.2 -> 0.38.3

mweinelt committed:
python3Packages.sumo: 2.3.0 -> 2.3.2

mweinelt committed:
python3Packages.testfixtures: 6.18.5 -> 7.0.0

mweinelt committed:
python3Packages.taskw: 1.3.1 -> 2.0.0

mweinelt committed:
python3Packages.tensorflow-datasets: 4.5.2 -> 4.6.0

mweinelt committed:
python3Packages.terminado: 0.13.3 -> 0.15.0

mweinelt committed:
python3Packages.tensorboard_plugin_profile: 2.5.0 -> 2.8.0

mweinelt committed:
python3Packages.tern: 2.10.0 -> 2.10.1

mweinelt committed:
python3Packages.tensorboard: 2.9.0 -> 2.9.1

mweinelt committed:
python3Packages.tensorboardx: 2.4 -> 2.5

mweinelt committed:
python3Packages.tempora: 5.0.1 -> 5.0.2

mweinelt committed:
python3Packages.torchmetrics: 0.8.2 -> 0.9.2

mweinelt committed:
python3Packages.timetagger: 22.6.4 -> 22.6.6

mweinelt committed:
python3Packages.tomlkit: 0.10.1 -> 0.11.1

mweinelt committed:
python3Packages.thinc: 8.0.17 -> 8.1.0

mweinelt committed:
python3Packages.tornado: 6.1 -> 6.2

mweinelt committed:
python3Packages.toolz: 0.11.2 -> 0.12.0

mweinelt committed:
python3Packages.tiledb: 0.13.0 -> 0.16.3

mweinelt committed:
python3Packages.torchinfo: 1.7.0 -> 1.64

mweinelt committed:
python3Packages.torchvision: 0.11.3 -> 0.13.0

mweinelt committed:
python3Packages.tox: 3.24.5 -> 3.25.1

mweinelt committed:
python3Packages.trezor_agent: 0.11.0 -> 0.12.0

mweinelt committed:
python3Packages.trimesh: 3.12.7 -> 3.12.8

mweinelt committed:
python3Packages.tpm2-pytss: 1.1.0 -> 1.2.0

mweinelt committed:
python3Packages.trio: 0.20.0 -> 0.21.0

mweinelt committed:
python3Packages.translatepy: 2.2 -> 2.3

mweinelt committed:
python3Packages.transformers: 4.19.4 -> 4.20.1

mweinelt committed:
python3Packages.ttp: 0.8.4 -> 0.9.1

mweinelt committed:
python3Packages.trytond: 6.4.2 -> 6.4.3

mweinelt committed:
python3Packages.types-redis: 4.3.3 -> 4.3.4

mweinelt committed:
python3Packages.types-pyyaml: 6.0.9 -> 6.0.10

mweinelt committed:
python3Packages.types-python-dateutil: 2.8.16 -> 2.8.18

mweinelt committed:
python3Packages.tubes: 0.2.0 -> 0.2.1

mweinelt committed:
python3Packages.typing-extensions: 4.1.1 -> 4.3.0

mweinelt committed:
python3Packages.ttp-templates: 0.1.4 -> 0.3.1

mweinelt committed:
python3Packages.typer: 0.4.2 -> 0.6.1

mweinelt committed:
python3Packages.types-protobuf: 3.19.15 -> 3.19.22

mweinelt committed:
python3Packages.typed-settings: 1.0.1 -> 1.1.0

mweinelt committed:
python3Packages.ufo2ft: 2.27.0 -> 2.28.0

mweinelt committed:
python3Packages.virtualenv: 20.14.0 -> 20.15.1

mweinelt committed:
python3Packages.uncertainties: 3.1.6 -> 3.1.7

mweinelt committed:
python3Packages.vcrpy: 4.1.1 -> 4.2.0

mweinelt committed:
python3Packages.untangle: 1.1.1 -> 1.2.1

mweinelt committed:
python3Packages.uproot: 4.2.0 -> 4.3.3

mweinelt committed:
python3Packages.ujson: 5.3.0 -> 5.4.0

mweinelt committed:
python3Packages.urllib3: 1.26.9 -> 1.26.10

mweinelt committed:
python3Packages.validators: 0.18.2 -> 0.20.0

mweinelt committed:
python3Packages.vivisect: 1.0.7 -> 1.0.8

mweinelt committed:
python3Packages.watchfiles: 0.14.1 -> 0.15.0

mweinelt committed:
python3Packages.watchdog: 2.1.8 -> 2.1.9

mweinelt committed:
python3Packages.watchgod: 0.8.1 -> 0.8.2

mweinelt committed:
python3Packages.vulture: 2.4 -> 2.5

mweinelt committed:
python3Packages.whisper: 1.1.8 -> 1.1.10

mweinelt committed:
python3Packages.websocket-client: 1.3.2 -> 1.3.3

mweinelt committed:
python3Packages.wandb: 0.12.20 -> 0.12.21

mweinelt committed:
python3Packages.wavedrom: 2.0.3.post2 -> 2.0.3.post3

mweinelt committed:
python3Packages.widgetsnbextension: 3.6.0 -> 3.6.1

mweinelt committed:
python3Packages.warlock: 1.3.3 -> 2.0.1

mweinelt committed:
python3Packages.xml2rfc: 3.12.4 -> 3.13.0

mweinelt committed:
python3Packages.wrf-python: 1.3.3 -> 1.3.4.1

mweinelt committed:
python3Packages.zict: 2.1.0 -> 2.2.0

mweinelt committed:
python3Packages.yamllint: 1.26.3 -> 1.27.1

mweinelt committed:
python3Packages.xarray-einstats: 0.2.2 -> 0.3.0

mweinelt committed:
python3Packages.wled: 0.13.2 -> 0.14.1

mweinelt committed:
python3Packages.zarr: 2.11.1 -> 2.12.0

mweinelt committed:
python3Packages.zope.configuration: 4.4.0 -> 4.4.1

mweinelt committed:
python3Packages.zstandard: 0.17.0 -> 0.18.0

mweinelt committed:
python3Packages.zope.testrunner: 5.4.0 -> 5.5

mweinelt committed:
python3Packages.zstd: 1.5.1.0 -> 1.5.2.5

risicle commented:
https://github.com/numba/numba/issues/8242#issuecomment-1184960325

> Neither Numba 0.55.2 and the current RC 0.56.0rc1 support NumPy 1.23. NumPy 1.23 support is planned for the next release (0.57).

mweinelt committed:
python3Packages.hatch-vcs: disable failing test

This test is incompatible with setuptools-scm>=7.0, but does not seem to
cause other problems as the upstream PR fixing this is also only
targeting the testsuite.

mweinelt committed:
python3Packages.s3transfer: 0.5.0 -> 0.6.0

mweinelt committed:
awscli: 1.22.88 -> 1.25.31

mweinelt committed:
libcamera: 2022-01-03 -> 2022-07-15

Fixes the documentation build with sphinx 5.0

mweinelt committed:
ansible-doctor: relax python-json-logger constraint

mweinelt committed:
python3Packages.agate-sql: reenable test and use pytest instead of nose

mweinelt committed:
python3Packages.fakeredis: update src hash

Failed to build on hydra due to hash mismatch.

mweinelt committed:
atomic-operator: fix src hash

Fails to build on hydra due to src hash mismatch.

mweinelt committed:
html-tidy: fix src hash

Fails to build on hydra due a src hash mismatch.

mweinelt committed:
rabbitvcs: fix src hash

Fails to build on hydra due a src hash mismatch.

SuperSandro2000 committed:
python310Packages.dbus: cleanup, add empty meta.maintainers

SuperSandro2000 committed:
python310Packages.python-dbusmock: 0.28.1 -> 0.28.2

mweinelt commented on pkgs/development/python-modules/ddt/default.nix:
```diff
@@ -16,16 +16,21 @@ buildPythonPackage rec {
   meta = with lib; {
     description = "Data-Driven/Decorated Tests, a library to multiply test cases";
     homepage = "https://github.com/txels/ddt";
+    maintainer = with maintainer; [ ];
```
maintainers

risicle commented:
Hmm `thrift` appears to be failing on macos, will have to look another time though

jonringer committed:
python3Packages.azure-core: disable transport tests

SuperSandro2000 committed:
python3Packages.geopy: fix geographiclib dependency

mweinelt committed:
python3Packages.openapi-spec-validator: fix build & tests

SuperSandro2000 committed:
python310Packages.pylama: move tests to passthru

mweinelt committed:
python3Packages.sphinxHook: run post install

This allows importing and inspecting the build result, which is required
for the psycopg (3.x) package.

SuperSandro2000 committed:
platformio: 6.0.1 -> 6.0.2

mweinelt committed:
python3Packages.psycopg: init at 3.0.15

There are two more variants that should be packaged, one that uses
Cython, and one that supports pooled connections. This package is the
pure python version.

Getting the docs to build required some work upstream, which is the
reason for a bunch of the patching we currently carry. I expect this to
go away over time, given that psycopg is very new-ish.

mweinelt committed:
python3Packages.pgspecial: psycopg2 out, psycopg in

mweinelt committed:
python3Packages.sphinx-autodoc-typehints: init at 1.18.3

Required for building the psycopg documentation.

SuperSandro2000 committed:
python310Packages.nbxmpp: fix pythonImportsCheck failing

SuperSandro2000 committed:
python310Packages.gtimelog: remove redundant meta.platforms

SuperSandro2000 committed:
python310Packages.pydbus: add pythonImportsCheck

SuperSandro2000 committed:
python310Packages.python-dbusmock: 0.28.2 -> 0.28.3

SuperSandro2000 committed:
python310Packages.pylint: 2.14.4 -> 2.14.5

SuperSandro2000 committed:
python310Packages.blinker: 1.4 -> 1.5

SuperSandro2000 committed:
python310Packages.limits: 2.6.3 -> 2.7.0

SuperSandro2000 committed:
python310Packages.dj-rest-auth: 2.2.4 -> 2.2.5

kirillrdy committed:
awsebcli: fixup, use more versions from nixpkgs

SuperSandro2000 commented on pkgs/development/interpreters/python/hooks/sphinx-hook.sh:
```diff
@@ -53,5 +53,5 @@ installSphinxPhase() {
 }
 
-preBuildPhases+=" buildSphinxPhase"
+preDistPhases+=" buildSphinxPhase"
```
Should we also rename ``buildSphinxPhase``?

mweinelt committed:
python3Packages.async-upnp-client: disable network tests

jonringer commented:
@mweinelt can you please add the link to the regression report to the PR description? :)

jonringer committed:
python3Packages.numba: bump supported numpy version

jonringer committed:
aws-sam-cli: fix deps

jonringer committed:
awscli2: fix build

jonringer committed:
calibre-web: fix deps

jonringer committed:
python3Packages.mdx-truly-sane-lists: fix build with markdown>=3.4

SomeoneSerge commented:
`cudaSupport = true;` run: https://hercules-ci.com/github/SomeoneSerge/nixpkgs-unfree/jobs/1302

CC @NixOS/cuda-maintainers

risicle commented:
setuptools >= 61.1 and its new patch is what thrift doesn't like I think.

mweinelt committed:
python3Packages.respx: run tests in legacy asyncio-mode

Since pytest-asyncio==0.19.0 the strict mode has become the default. The
respx module has not been upgraded with that in mind, so let's rollback
that change to fix the tests.

risicle commented:
https://github.com/risicle/thrift/commit/d34d67cabe343fab609b477dba51f05c71c952af gets us half the way with thrift, will figure out the rest and send it upstream tomorrow.

SuperSandro2000 committed:
python310Packages.pyqt: fix build by confirming the license

mweinelt committed:
home-assistant: relax ifaddr constraint

risicle commented:
https://github.com/apache/thrift/pull/2635

risicle committed:
thrift: add patch fixing tests with setuptools >= 62.1.0

mweinelt committed:
python3Packages.enrich: disable failing test

mweinelt committed:
python3Packages.celery: patch billiard>=4.0 compat

mweinelt committed:
python3Packages.siosocks: disable network tests

mweinelt committed:
python3Packages.structlog: reset asyncio test mode to legacy

mweinelt committed:
python3Packages.markdown-include: 0.6.0 -> 0.7.0

mweinelt committed:
python3Packages.ansible-later: fix build

SuperSandro2000 committed:
backblaze-b2: fix tests, little cleanup

SuperSandro2000 committed:
python310Packages.safeio: init at 1.2

SuperSandro2000 committed:
swaglyrics: fix dependencies

mweinelt committed:
python3Packages.flask_sqlalchemy: disable flaky test

mweinelt committed:
python3Packages.flask-sqlalchemy: rename from flask_sqlalchemy

SuperSandro2000 committed:
python310Packages.basemap-data: init at 1.3.3

SuperSandro2000 committed:
fixup! python310Packages.pyshp: 2.2.0 -> 2.3.0

SuperSandro2000 committed:
python310Packages.textx: use python.sitePackages instead of constructing it manully

SuperSandro2000 committed:
atomic-operator: fix dependencies

SuperSandro2000 committed:
ioccheck: fix one dependency

SuperSandro2000 committed:
khal: disable timing based test

mweinelt committed:
python3Packages.subprocess-tee: disable rich test

SuperSandro2000 committed:
python310Packages.eth-abi: ignore flaky test on hydra

mweinelt committed:
python3Packages.aioshelly: 2.0.0 -> 2.0.1

mweinelt committed:
python3Packages.pyairvisual: 2021.10.0 -> 2022.07.0

mweinelt committed:
python3Packages.pytomorrowio: 0.3.3 -> 0.3.4

mweinelt committed:
python3Packages.dynalite-devices: test in asyncio legacy mode

mweinelt committed:
python3Packages.pypck: test in asyncio legacy mode

mweinelt committed:
python3Packages.pyotgw: 2021-03-25 -> 2.0.0

mweinelt committed:
python3Packages.pywizlight: test in asyncio legacy mode

mweinelt committed:
python3Packages.tifffile: disable failing test

mweinelt committed:
python3Packages.sanic: disable failing test

mweinelt committed:
python3Packages.channels: test in asyncio legacy mode

mweinelt committed:
python3Packages.amqtt: 2022-01-11 -> 202205-29

mweinelt committed:
python3Packages.jsonrpc-websocket: test in asyncio legacy mode

mweinelt committed:
python3Packages.zigpy-znp: test in asyncio legacy mode

mweinelt committed:
python3Packages.prance: relax chardet constraint

mweinelt committed:
python3Packages.hypercorn: test in asyncio legacy mode

mweinelt committed:
python3Packages.aioftp: test in asyncio legacy mode

mweinelt committed:
python3Packages.pyatv: test in asyncio legacy mode

SuperSandro2000 committed:
python310Packages.ihatemoney: remove unused input

mweinelt committed:
python3Packages.homematicip: test in asyncio legacy mode

mweinelt committed:
python3Packages.pylutron-caseta: test in asyncio legacy mode

mweinelt committed:
python3Packages.renault-api: test in asyncio legacy mode

mweinelt committed:
home-assistant: pin backoff before 2.0 bump

mweinelt committed:
python3Packages.python-smarttub: test in asyncio legacy mode

mweinelt committed:
python3Packages.pyspcwebgw: test in asyncio legacy mode

mweinelt committed:
python3Packages.discovery30303: test in asyncio legacy mode

mweinelt committed:
python3Packages.aioswitcher: test in asyncio legacy mode

mweinelt committed:
python3Packages.aiosyncthing: test in asyncio legacy mode

mweinelt committed:
python3Packages.pytautulli: test in asyncio legacy mode

mweinelt committed:
python3Packages.python-kasa: test in asyncio legacy mode

mweinelt committed:
python3Packages.xbox-webapi: test in asyncio legacy mode

mweinelt committed:
python3Packages.sphinx-jupyterbook-latex: relax sphinx constraint

mweinelt committed:
python3Packages.mailsuite: 1.8.0 -> 1.9.2

mweinelt committed:
python3Packages.snowflake-connector-python: relax charset-normalizer constraint

mweinelt committed:
python3Packages.awkward: skip failing test

collares committed:
python3Packages.rpy2: update R_LIBS_SITE patch

mweinelt committed:
python3Packages.cattrs: 1.10.0 -> 22.1.0

https://github.com/python-attrs/cattrs/blob/main/HISTORY.rst#2210-2022-04-03

mweinelt committed:
maturin: 0.12.9 -> 0.13.0

mweinelt committed:
python3Packages.subarulink: test in asyncio legacy mode

mweinelt committed:
python3Packages.orjson: 3.6.7 -> 3.7.8

mweinelt committed:
python3Packages.unifi-discovery: test in asyncio legacy mode

mweinelt committed:
python3Packages.qiskit-terra: 0.20.1 -> 0.21.0

collares committed:
python3Packages.sphinx: do not use deprecated docutils 0.19 API

mweinelt committed:
python3Packages.requests-cache: 0.9.4 -> 0.9.5

K900 committed:
python3Packages.pyqt-builder: use sip-distinfo from PATH

Fixes PyQt5 build.

K900 committed:
python3Packages.pyqt5: 5.15.4 -> 5.15.7

risicle committed:
python3Packages.typer: include rich in checkInputs

risicle committed:
python3Packages.pyosmium: switch to pytestCheckHook

risicle committed:
python3Packages.islpy: 2022.2 -> 2022.2.1

risicle committed:
python3Packages.owslib: remove pyproj version constraint

risicle committed:
python3Packages.pyquil: remove qcs-api-client version constraint

as already done upstream https://github.com/rigetti/pyquil/pull/1449

mweinelt committed:
python3Packages.matrix-nio: run tests in asyncio legacy mode

mweinelt committed:
python3Packages.dask: 2022.05.2 -> 2022.7.0
